### PR TITLE
fix(selector): root classname is applied twice

### LINKED ISSUE #2396: numericSelector cssClasses option for root applies to a parent div as well as the child <select>
```js
search.addWidget(
    instantsearch.widgets.numericSelector({
        container: '#numericSelector',
        attributeName: 'thing.id',
        options: [{label: 'name of thing', value: 1}],
        cssClasses: {
            root: 'form-control'
        }
    })
);
```

outputs the following HTML:
```html
<div>
    <div class="ais-root ais-numeric-selector form-control">
        <div class="ais-body">
            <select class="ais-numeric-selector form-control">
                <option class="ais-numeric-selector--item" value="1">name of thing</option>
            </select>
        </div>
    </div>
</div>
```

According to the documentation, it should only be applied to the select element. This is problematic in this scenario because I am using with bootstrap and it is making the output undesirable

### PR DESCRIPTION
**Pull request by bobylito**
The implementation of the React component `Selector` was applying the `root` css
classes that were already applied by the headerfooter. In order to
prevent regressions and let user customize the <select>, this patch adds
a new key to customize cssClasses in Selector bases widgets (`select`), and this
new key has the root value to avoid breaking the current implementations.

Fix #2396 #2397

### CODE REVIEW

**Comment by bobylito** [comment]
I need push them 🤣

**Comment by bobylito** [comment]
Good thing that we have argos :D

**Comment by bobylito** [comment]
The select was used but not implemented. The change is expected on argos.

File: src/components/__tests__/__snapshots__/Selector-test.js.snap [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 exports[`Selector should render <Selector/> with numbers 1`] = `
 <select
-  className="custom-root"
+  className={undefined}
```
Comment (vvo) [inline src/components/__tests__/__snapshots__/Selector-test.js.snap]: Should we update this test to not have undefined inside it? And check if select css class is actually read?

File: src/widgets/sort-by-selector/sort-by-selector.js [hunk 2]
```diff
@@ -31,13 +31,14 @@ const usage = `Usage:
...
   indices,
-  [cssClasses.{root,item}={}],
+  [cssClasses.{root,select,item}={}],
   [autoHideContainer=false]
 })`;
 
 /**
  * @typedef {Object} SortByWidgetCssClasses
- * @property {string|string[]} [root] CSS classes added to the parent `<select>`.
+ * @property {string|string[]} [root] CSS classes added to the outer `<div>`.
```
Comment (vvo) [inline src/widgets/sort-by-selector/sort-by-selector.js]: 👍

**Pull request merged**